### PR TITLE
Generated resource and datasource for fvSiteAssociated and fvRemoteId

### LINKED ISSUE #1224: Add resource and datasource for fvSiteAssociated and fvRemoteId (DCNE-190)
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### New or Affected Resource(s) + ACI Class(es):

* aci_associated_site + fv:SiteAssociated
* aci_remote_site_id_mappings + fv:RemoteId

### APIC version and APIC Platform

* V x.x.x and on-prem/cloud-aws/cloud-azure/all.

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* #0000

### PR DESCRIPTION
**Pull request by abrahammughal**
Fixes #1224

### CODE REVIEW

File: docs/resources/associated_site.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+* Location: `Not shown in UI`
+
+## Example Usage ##
+
+The configuration snippet below creates a Associated Site with only required attributes.
+
+```hcl
+
+resource "aci_associated_site" "example_application_epg" {
+  parent_dn = aci_application_epg.example.id
```
Comment (lhercot) [inline docs/resources/associated_site.md:42]: I think side_id should be mandatory

File: docs/resources/associated_site.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+resource "aci_associated_site" "full_example_vrf" {
+  parent_dn   = aci_vrf.example.id
+  annotation  = "annotation"
+  description = "description"
+  name        = "name"
+  name_alias  = "name_alias"
+  owner_key   = "owner_key"
+  owner_tag   = "owner_tag"
+  site_id     = "0.000000"
```
Comment (lhercot) [inline docs/resources/associated_site.md:87]: this should be an integer

File: docs/resources/associated_site.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+```
+terraform import aci_associated_site.example_application_epg uni/tn-{name}/mscGraphXlateCont/epgDefXlate-[{epgDefDn}]/stAsc
+```
+
+Starting in Terraform version 1.5, an existing Associated Site can be imported
+using [import blocks](https://developer.hashicorp.com/terraform/language/import) via the following configuration:
+
+```
+import {
+  id = "uni/tn-{name}/mscGraphXlateCont/epgDefXlate-[{epgDefDn}]/stAsc"
```
Comment (lhercot) [inline docs/resources/associated_site.md:162]: Anyway we can use a different example than this one?

**Review by lhercot — CHANGES_REQUESTED** [review]

File: docs/data-sources/associated_site.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+# aci_associated_site #
+
+Data source for Associated Site
+
+## API Information ##
+
+* Class: [fvSiteAssociated](https://pubhub.devnetcloud.com/media/model-doc-latest/docs/app/index.html#/objects/fvSiteAssociated/overview)
+
+
+* Distinguished Name Format: `uni/tn-{name}/ctx-{name}/stAsc`
```
Comment (akinross) [inline docs/data-sources/associated_site.md:22]: is this not supporting more dn_formats? please add to the classes.yml those dn_formats that should be added

File: docs/data-sources/associated_site.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+## Schema ##
+
+### Required ###
+
+* `parent_dn` - (string) The distinguished name (DN) of the parent object, possible resources:
+  - [aci_application_epg](https://registry.terraform.io/providers/CiscoDevNet/aci/latest/docs/resources/application_epg) ([fvAEPg](https://pubhub.devnetcloud.com/media/model-doc-latest/docs/app/index.html#/objects/fvAEPg/overview))
+  - [aci_vrf](https://registry.terraform.io/providers/CiscoDevNet/aci/latest/docs/resources/vrf) ([fvCtx](https://pubhub.devnetcloud.com/media/model-doc-latest/docs/app/index.html#/objects/fvCtx/overview))
+  - The distinquised name (DN) of classes below can be used but currently there is no available resource for it:
+    - [fvBD](https://pubhub.devnetcloud.com/media/model-doc-latest/docs/app/index.html#/objects/fvBD/overview)
+    - [l3extInstP](https://pubhub.devnetcloud.com/media/model-doc-latest/docs/app/index.html#/objects/l3extInstP/overview)
```
Comment (akinross) [inline docs/data-sources/associated_site.md:51]: We have resources for these classes. Please lookup the resource name for these classes in the legacy resource (/aci) directory. Then add them to the classes.yaml like followed:

```
fvBD:
  resource_name: "bridge_domain"

File: docs/data-sources/associated_site.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+---
+
+# aci_associated_site #
+
+Data source for Associated Site
+
+## API Information ##
+
+* Class: [fvSiteAssociated](https://pubhub.devnetcloud.com/media/model-doc-latest/docs/app/index.html#/objects/fvSiteAssociated/overview)
+
```
Comment (akinross) [inline docs/data-sources/associated_site.md:20]: missing versions, please check you used the meta file from previously shared link that includes versions

File: docs/data-sources/remote_site_id_mappings.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+```
+
+## Schema ##
+
+### Required ###
+
+* `parent_dn` - (string) The distinguished name (DN) of the parent object, possible resources:
+  - [aci_associated_site](https://registry.terraform.io/providers/CiscoDevNet/aci/latest/docs/resources/associated_site) ([fvSiteAssociated](https://pubhub.devnetcloud.com/media/model-doc-latest/docs/app/index.html#/objects/fvSiteAssociated/overview))
+* `site_id` (siteId) - (string) Remote Site ID.
```
Comment (akinross) [inline docs/data-sources/remote_site_id_mappings.md:46]: Should we add some more explanation about what this site id actually is and where / how users should decide the input?

File: docs/data-sources/associated_site.md (hunk 6 above)
Comment (akinross) [reply docs/data-sources/associated_site.md:20]: all the documentation comments apply to the resource equivalent docs also

File: docs/resources/remote_site_id_mappings.md [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+
+resource "aci_remote_site_id_mappings" "full_example_associated_site" {
+  parent_dn         = aci_associated_site.example.id
+  annotation        = "annotation"
+  description       = "description"
+  name              = "name"
+  name_alias        = "name_alias"
+  owner_key         = "owner_key"
+  owner_tag         = "owner_tag"
+  remote_ctx_pc_tag = "remote_ctx_pc_tag"
```
Comment (akinross) [inline docs/resources/remote_site_id_mappings.md:57]: is ctx referring to vr? should we rename this naming?

File: docs/resources/remote_site_id_mappings.md [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+### Optional ###
+  
+* `annotation` (annotation) - (string) The annotation of the Remote Site Id Mappings object.
+  - Default: `orchestrator:terraform`
+* `description` (descr) - (string) The description of the Remote Site Id Mappings object.
+* `name` (name) - (string) The name of the Remote Site Id Mappings object.
+* `name_alias` (nameAlias) - (string) The name alias of the Remote Site Id Mappings object.
+* `owner_key` (ownerKey) - (string) The key for enabling clients to own their data for entity correlation.
+* `owner_tag` (ownerTag) - (string) A tag for enabling clients to add their own data. For example, to indicate who created this object.
+* `remote_ctx_pc_tag` (remoteCtxPcTag) - (string) Remote context's *pcTag*, mapping required for default route case.
```
Comment (akinross) [inline docs/resources/remote_site_id_mappings.md:99]: same context rename?

File: gen/testvars/fvRemoteId.yaml [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+
+all:
+  annotation: "annotation"
+  description: "description"
+  name: "name"
+  name_alias: "name_alias"
+  owner_key: "owner_key"
+  owner_tag: "owner_tag"
+  remote_ctx_pc_tag: "remote_ctx_pc_tag"
+  remote_pc_tag: "remote_pc_tag"
```
Comment (akinross) [inline gen/testvars/fvRemoteId.yaml:33]: is apic allowing just regular strings for this? would have expected a pctag id here

File: gen/testvars/fvRemoteId.yaml [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+  site_id: "0"
```
Comment (akinross) [inline gen/testvars/fvRemoteId.yaml:34]: can we change the site_id during update?

**Review by akinross — CHANGES_REQUESTED** [review]

File: gen/testvars/fvRemoteId.yaml (hunk 11 above)
Comment (abrahammughal) [reply gen/testvars/fvRemoteId.yaml:34]: Yes

**Pull request closed without merging**